engine/src/main/java/org/hibernate/validator/constraints/SafeHtml.java

```diff
@@ -84,6 +91,29 @@
 	}
 
 	/**
+	 * Allows to specify whitelist protocols within specified tag as a value of a specified key.
+	 */
+	@Target({ METHOD, FIELD, ANNOTATION_TYPE, CONSTRUCTOR, PARAMETER })
+	@Retention(RUNTIME)
+	@Documented
+	public @interface Protocol {
+		/**
+		 * @return the tag the URL protocol is for.
+		 */
+		String tag();
+
+		/**
+		 * @return the attribute key.
+		 */
+		String key();
```

Comment:
Should we decide to go this route with a new annotation, it should be named `attribute()` for the sake of consistency. Also some `@since` tags on the new annotation and on `additionalProtocols()` would be good.